correctness/test_correctness.py

```diff
@@ -0,0 +1,350 @@
+    
+    For existing DBs, we wait 3 seconds and query with [3s] to avoid results from previous runs.
+    """
+    if FRESH_TSDB:
+        # Test empty on initialization
+        query = 'prometheus_http_requests_total{}'
+        read_response = mockmetheus_instance.remote_read(query)
+        assert(is_empty(read_response))
+    else:
+        # Test empty for existing TSDB
+        time.sleep(3)
+        query = 'prometheus_http_requests_total{}[3s]'
+        read_response = mockmetheus_instance.remote_read(query)
+        assert(is_empty(read_response))
+
+def test_read_metric_dne(mockmetheus_instance):
+    """
+    Remote read with a DNE metric
+    """
+    read_response = mockmetheus_instance.remote_read('non_existent_metric')
+    assert(is_empty(read_response))
+
+def test_write_no_data(mockmetheus_instance):
+    """
+    Remote write with an empty timeseries
+    """
+    timeseries_data = []
+    write_response = mockmetheus_instance.remote_write(timeseries_data)
+    assert(write_response == 200)
+
+def test_write_no_labels(mockmetheus_instance):
+    """
+    Remote write with no labels
+    """
+    timeseries_data = [
+        {
+            'labels': {},
+            'samples': [
+                (220, int(time.time() * 1000)),
+            ]
+        }
+    ]
+    try:
+        write_response = mockmetheus_instance.remote_write(timeseries_data)
+    except Exception as e:
+        assert e.response.status_code == 400
+
+def test_write_no_samples(mockmetheus_instance):
+    """
+    Remote write & read with no samples
+    """
+    __name__ = 'prometheus_http_requests_total'
+    instance = 'mockmetheus'
+    timeseries_data = [
+        {
+            'labels': {
+                '__name__': __name__,
+                'instance': instance
+            },
+            'samples': []
+        }
+    ]
+    write_response = mockmetheus_instance.remote_write(timeseries_data)
+    assert(write_response == 200)
+
+    # ensures data is ingested into Timestream
+    time.sleep(1)
+
+    query = f'prometheus_http_requests_total{{instance="{instance}"}}[3s]'
+    read_response = mockmetheus_instance.remote_read(query)
+    assert(is_empty(read_response))
+
+def test_success(mockmetheus_instance):
+    """
+    Remote write & read with valid data
+    """
+    __name__ = 'prometheus_http_requests_total'
+    instance = 'mockmetheus'
+    expected_sample_record = 220
+    test_id = generate_test_run_id()
+
+    timeseries_data = [
+        {
+            'labels': {
+                '__name__': __name__,
+                'instance': instance,
+                'test_id': test_id
+            },
+            'samples': [
+                (expected_sample_record, int(time.time() * 1000)),
+            ]
+        }
+    ]
+    write_response = mockmetheus_instance.remote_write(timeseries_data)
+    assert(write_response == 200)
+
+    # ensures data is ingested into Timestream
+    time.sleep(1)
```

Comment:
For this and the other sleeps, can network latency have any impact on whether data is finished ingesting into Timestream?

Reply:
Making it configurable is a good idea. I think using a named constant, like `SLEEP_TIME`, would be good enough.